js/wp-seo-post-scraper-305.js

```diff
@@ -183,13 +183,17 @@
 	/**
 	 * Gets content from the content field, if tinyMCE is initialized, use the getContent function to get the data from tinyMCE
 	 * If tiny is hidden, take the value from the contentfield, since tinyMCE isn't updated when it isn't visible.
+	 * TinyMCE doesn't update the content field realtime, it has a delay in updating this field, so getting the value of the content
+	 * field isn't always reliable.
 	 * @returns {String}
 	 */
 	PostScraper.prototype.getContentTinyMCE = function() {
 		var val = document.getElementById( 'content' ) && document.getElementById( 'content' ).value || '';
```

Comment:
The points that @jcomack has brought up are valid points, but they point to an underlying problem. The code in this function is very unstructured.

You should add a new function `getTinyMCEElementContent` that retrieves the content from the element with `document.getElementById` and `.value` and then rewrite the current function with an early return like this:

```js
if ( typeof tinyMCE === "undefined" || typeof tinyMCE.editors === "undefined" || tinyMCE.editors.length === 0 ) {
    return this.getTinyMCEElementContent();
}
```